### PR TITLE
[16.0][MIG] purchase_order_secondary_unit: Migration to 16.0

### PR DESCRIPTION
**Pull request by acsonefho**
Migration of `purchase_order_secondary_unit`.

Based on closed branch: https://github.com/OCA/purchase-workflow/pull/1970

### CODE REVIEW

File: purchase_order_secondary_unit/__manifest__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 {
     "name": "Purchase Order Secondary Unit",
     "summary": "Purchase product in a secondary unit",
-    "version": "15.0.1.1.0",
+    "version": "16.0.1.1.0",
```
Comment (SirAionTech) [inline purchase_order_secondary_unit/__manifest__.py:6]: Shouldn't we start from `16.0.1.0.0`?

File: purchase_order_secondary_unit/models/purchase_order.py [hunk 2]
```diff
@@ -12,21 +12,30 @@ class PurchaseOrderLine(models.Model):
     }
 
     product_qty = fields.Float(
-        store=True, readonly=False, compute="_compute_product_qty", copy=True
+        store=True,
+        readonly=False,
+        compute="_compute_product_qty",
```
Comment (SirAionTech) [inline purchase_order_secondary_unit/models/purchase_order.py:17]: Most of these attributes are redundant because already in https://github.com/odoo/odoo/blob/d59e645b2edf167cb7f108fd7db563799b3db2f1/addons/purchase/models/purchase.py#L940-L941, can we only override the field attributes that need to be changed?

File: purchase_order_secondary_unit/models/purchase_order.py [hunk 3]
```diff
@@ -12,21 +12,30 @@ class PurchaseOrderLine(models.Model):
...
+    )
+    product_packaging_qty = fields.Float(
+        compute="_compute_product_packaging_qty", store=True, precompute=True
+    )
+    product_packaging_id = fields.Many2one(
+        compute="_compute_product_packaging_id", store=True, precompute=True
     )
 
-    @api.depends("secondary_uom_qty", "secondary_uom_id")
+    @api.depends("product_packaging_qty", "secondary_uom_qty", "secondary_uom_id")
```
Comment (SirAionTech) [inline purchase_order_secondary_unit/models/purchase_order.py:28]: Fields in `api.depends` are incremental (see https://github.com/odoo/odoo/blob/68c8d5012700d04ef253d22be3440b9922f1d8aa/odoo/fields.py#L577), this means that you only need to add the new fields your override needs to depend on, so you don't need to add `product_packaging_qty` here.

File: purchase_order_secondary_unit/models/purchase_order.py [hunk 4]
```diff
@@ -12,21 +12,30 @@ class PurchaseOrderLine(models.Model):
...
+        return super()._compute_product_qty()
 
     @api.onchange("product_uom")
-    def onchange_product_uom_for_secondary(self):
+    def _onchange_product_uom_for_secondary(self):
         self._onchange_helper_product_uom_for_secondary()
 
     @api.onchange("product_id")
-    def onchange_product_id(self):
+    def _onchange_product_id(self):
```
Comment (SirAionTech) [inline purchase_order_secondary_unit/models/purchase_order.py:38]: In `15.0` this was an override of `onchange_product_id` (https://github.com/odoo/odoo/blob/a56586119845969e9d867a220f5330a6c7daa5c2/addons/purchase/models/purchase.py#L1132); why in `16.0` this is no more an override of `onchange_product_id` (https://github.com/odoo/odoo/blob/d59e645b2edf167cb7f108fd7db563799b3db2f1/addons/purchase/models/purchase.py#L1174)?

File: purchase_order_secondary_unit/views/purchase_order_views.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                        attrs="{'readonly': [('state', 'in', ('done', 'cancel'))]}"
+                    />
+                <field
+                        name="secondary_uom_id"
+                        class="oe_inline"
+                        domain="['|', ('product_id', '=', product_id),
+                                '&amp;', ('product_tmpl_id.product_variant_ids', 'in', [product_id]),
+                                         ('product_id', '=', False)]"
+                        options="{'no_create': True}"
+                        attrs="{'readonly': [('state', 'in', ('done', 'cancel'))]}"
```
Comment (SirAionTech) [inline purchase_order_secondary_unit/views/purchase_order_views.xml:27]: We are missing https://github.com/OCA/purchase-workflow/pull/1995, could you please include it?

File: purchase_order_secondary_unit/models/purchase_order.py [hunk 6]
```diff
@@ -15,18 +15,17 @@ class PurchaseOrderLine(models.Model):
...
     )
 
-    @api.depends("secondary_uom_qty", "secondary_uom_id")
+    @api.depends("product_packaging_qty", "secondary_uom_qty", "secondary_uom_id")
     def _compute_product_qty(self):
-        if hasattr(super(), "_compute_product_qty"):
-            return super()._compute_product_qty()
-        return self._compute_helper_target_field_qty()
+        self._compute_helper_target_field_qty()
+        return super()._compute_product_qty()
```
Comment (SirAionTech) [inline purchase_order_secondary_unit/models/purchase_order.py]: As you stated in the commit message, this is a bug that affects `15.0` too: can you extract this change (and the related test changes) so it can be easily backported? Or at least create an issue so it is known that there is this bug in `15.0`.

File: purchase_order_secondary_unit/models/purchase_order.py [hunk 7]
```diff
@@ -15,18 +15,17 @@ class PurchaseOrderLine(models.Model):
...
 
     @api.onchange("product_uom")
-    def onchange_product_uom_for_secondary(self):
+    def _onchange_product_uom_for_secondary(self):
```
Comment (SirAionTech) [inline purchase_order_secondary_unit/models/purchase_order.py]: Could you please clarify why this change is needed for the migration?

**Review by SirAionTech — CHANGES_REQUESTED** [review]
Thanks for the PR!
Let me know what you think of the comments below.
Could you please also squash the bots commits as explained in https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate?

**Comment by francesco-ooops** [comment]
@acsonefho are you planning to complete this migration?

**Comment by josepmy** [comment]
@acsonefho any new about this migration? can I help?

**Comment by pedrobaeza** [comment]
Superseded by #2216
